autocompleter/base.py

```diff
@@ -560,17 +562,50 @@ def suggest(self, term):
             if len(term) < MIN_LETTERS:
                 continue
 
-            result_keys = []
+            term_result_keys = []
             for norm_term in norm_terms:
                 norm_words = norm_term.split()
-                term_result_key = intermediate_result_key + '.' + norm_term
-                result_keys.append(term_result_key)
+                term_result_key = base_term_result_key + '.' + norm_term
+                term_result_keys.append(term_result_key)
                 keys = [PREFIX_BASE_NAME % (provider_name, norm_word,) for norm_word in norm_words]
                 pipe.zinterstore(term_result_key, keys, aggregate='MIN')
-            pipe.zunionstore(intermediate_result_key, result_keys, aggregate='MIN')
-            for term_result_key in result_keys:
+            pipe.zunionstore(base_term_result_key, term_result_keys, aggregate='MIN')
+            for term_result_key in term_result_keys:
                 pipe.delete(term_result_key)
-            pipe.zrange(intermediate_result_key, 0, MAX_RESULTS - 1)
+
+            if len(facets) > 0:
+                facet_result_keys = []
+                for facet in facets:
+                    try:
+                        facet_type = facet['type']
+                        if facet_type not in ['and', 'or']:
+                            continue
+                        facet_list = facet['facets']
+                        facet_set_keys = []
+                        for facet_dict in facet_list:
+                            facet_set_key = FACET_SET_BASE_NAME % (provider_name, facet_dict['key'], facet_dict['value'],)
+                            facet_set_keys.append(facet_set_key)
+                        facet_result_key = RESULT_SET_BASE_NAME % str(uuid.uuid4())
+                        if facet_type == 'and':
+                            pipe.zinterstore(facet_result_key, facet_set_keys, aggregate='MIN')
+                        else:
+                            pipe.zunionstore(facet_result_key, facet_set_keys, aggregate='MIN')
+                        facet_result_keys.append(facet_result_key)
+                    except KeyError:
+                        continue
+
+                # We want to calculate the intersection of all the intermediate facet sets created so far
+                # along with the temp term result set. So we need to use a new unique name for the
```

Comment:
ref to old names of vars "temp term result set" => "base term result set" and so on